PEP8

Code need PEP8 check / cleanup

We could add in `.travis.yml`

```
install:
  - pip install -qq flake8
  - .........

script:
    - nosetests ........
    - flake8 --version
    - flake8 --ignore E501 morse_talk

```

```
morse_talk/__init__.py:1:1: F403 'from .encoding import *' used; unable to detect undefined names
morse_talk/__init__.py:1:1: F401 '.encoding.*' imported but unused
morse_talk/__init__.py:2:1: F403 'from .decoding import *' used; unable to detect undefined names
morse_talk/__init__.py:2:1: F401 '.decoding.*' imported but unused
morse_talk/cli_mplot.py:19:1: F401 'morse_talk as mtalk' imported but unused
morse_talk/cli_mplot.py:24:1: E302 expected 2 blank lines, found 1
morse_talk/cli_mplot.py:41:5: F841 local variable '_' is assigned to but never used
morse_talk/cli_msound.py:23:21: E128 continuation line under-indented for visual indent
morse_talk/cli_msound.py:25:1: E302 expected 2 blank lines, found 1
morse_talk/cli_msound.py:51:28: E261 at least two spaces before inline comment
morse_talk/cli_msound.py:63:33: E261 at least two spaces before inline comment
morse_talk/cli_msound.py:66:14: E261 at least two spaces before inline comment
morse_talk/cli_mtree.py:19:1: E302 expected 2 blank lines, found 1
morse_talk/decoding.py:16:1: E302 expected 2 blank lines, found 1
morse_talk/decoding.py:40:25: E226 missing whitespace around arithmetic operator
morse_talk/decoding.py:40:35: E226 missing whitespace around arithmetic operator
morse_talk/decoding.py:41:29: E226 missing whitespace around arithmetic operator
morse_talk/decoding.py:41:39: E226 missing whitespace around arithmetic operator
morse_talk/decoding.py:45:28: E226 missing whitespace around arithmetic operator
morse_talk/decoding.py:45:42: E226 missing whitespace around arithmetic operator
morse_talk/decoding.py:54:51: E226 missing whitespace around arithmetic operator
morse_talk/decoding.py:54:70: E226 missing whitespace around arithmetic operator
morse_talk/encoding.py:12:1: E402 module level import not at top of file
morse_talk/encoding.py:65:1: E302 expected 2 blank lines, found 1
morse_talk/encoding.py:79:1: E302 expected 2 blank lines, found 1
morse_talk/encoding.py:90:1: E302 expected 2 blank lines, found 1
morse_talk/encoding.py:106:1: E302 expected 2 blank lines, found 1
morse_talk/encoding.py:118:17: E201 whitespace after '{'
morse_talk/encoding.py:118:38: E226 missing whitespace around arithmetic operator
morse_talk/encoding.py:120:54: E261 at least two spaces before inline comment
morse_talk/encoding.py:123:1: E302 expected 2 blank lines, found 1
morse_talk/encoding.py:139:1: E302 expected 2 blank lines, found 1
morse_talk/encoding.py:139:56: E251 unexpected spaces around keyword / parameter equals
morse_talk/encoding.py:139:58: E251 unexpected spaces around keyword / parameter equals
morse_talk/encoding.py:139:62: E226 missing whitespace around arithmetic operator
morse_talk/encoding.py:187:1: E302 expected 2 blank lines, found 1
morse_talk/plot.py:16:1: E302 expected 2 blank lines, found 1
morse_talk/plot.py:29:1: E302 expected 2 blank lines, found 1
morse_talk/plot.py:53:1: E302 expected 2 blank lines, found 1
morse_talk/plot.py:53:33: E251 unexpected spaces around keyword / parameter equals
morse_talk/plot.py:53:35: E251 unexpected spaces around keyword / parameter equals
morse_talk/plot.py:70:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:34:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:39:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:47:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:61:1: W293 blank line contains whitespace
morse_talk/sound.py:64:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:92:23: E701 multiple statements on one line (colon)
morse_talk/sound.py:93:23: E701 multiple statements on one line (colon)
morse_talk/sound.py:94:49: E261 at least two spaces before inline comment
morse_talk/sound.py:100:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:108:29: W291 trailing whitespace
morse_talk/sound.py:110:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:114:72: W291 trailing whitespace
morse_talk/sound.py:115:13: E128 continuation line under-indented for visual indent
morse_talk/sound.py:119:1: E302 expected 2 blank lines, found 1
morse_talk/sound.py:132:9: E128 continuation line under-indented for visual indent
morse_talk/sound.py:135:1: E302 expected 2 blank lines, found 1
morse_talk/tree.py:35:1: E302 expected 2 blank lines, found 1
morse_talk/tree.py:68:9: E265 block comment should start with '# '
morse_talk/tree.py:73:1: W293 blank line contains whitespace
morse_talk/tree.py:74:42: E251 unexpected spaces around keyword / parameter equals
morse_talk/tree.py:74:44: E251 unexpected spaces around keyword / parameter equals
morse_talk/tree.py:74:59: E251 unexpected spaces around keyword / parameter equals
morse_talk/tree.py:74:61: E251 unexpected spaces around keyword / parameter equals
morse_talk/tree.py:82:17: E265 block comment should start with '# '
morse_talk/tree.py:84:13: E265 block comment should start with '# '
morse_talk/tree.py:112:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:21:1: E402 module level import not at top of file
morse_talk/utils.py:23:1: E402 module level import not at top of file
morse_talk/utils.py:24:1: E402 module level import not at top of file
morse_talk/utils.py:26:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:42:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:58:15: E261 at least two spaces before inline comment
morse_talk/utils.py:61:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:63:43: W291 trailing whitespace
morse_talk/utils.py:112:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:114:63: W291 trailing whitespace
morse_talk/utils.py:145:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:154:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:159:34: E261 at least two spaces before inline comment
morse_talk/utils.py:161:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:165:12: W291 trailing whitespace
morse_talk/utils.py:178:9: E265 block comment should start with '# '
morse_talk/utils.py:179:9: E265 block comment should start with '# '
morse_talk/utils.py:181:25: E225 missing whitespace around operator
morse_talk/utils.py:187:25: E225 missing whitespace around operator
morse_talk/utils.py:192:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:200:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:209:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:217:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:241:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:263:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:282:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:300:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:322:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:340:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:353:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:355:12: W291 trailing whitespace
morse_talk/utils.py:380:1: E302 expected 2 blank lines, found 1
morse_talk/utils.py:383:1: W293 blank line contains whitespace
morse_talk/utils.py:393:21: E701 multiple statements on one line (colon)
morse_talk/utils.py:394:21: E701 multiple statements on one line (colon)
morse_talk/utils.py:399:1: E302 expected 2 blank lines, found 1
morse_talk/tests/test_encoding.py:2:1: F403 'from nose.tools import *' used; unable to detect undefined names
morse_talk/tests/test_encoding.py:5:1: E302 expected 2 blank lines, found 1
morse_talk/tests/test_encoding.py:12:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:13:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:14:13: E124 closing bracket does not match visual indentation
morse_talk/tests/test_encoding.py:15:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:22:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:23:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:24:13: E124 closing bracket does not match visual indentation
morse_talk/tests/test_encoding.py:25:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:28:5: E303 too many blank lines (2)
morse_talk/tests/test_encoding.py:32:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:38:17: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:39:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:45:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:46:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:47:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:48:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:49:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:55:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:56:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:58:1: E302 expected 2 blank lines, found 1
morse_talk/tests/test_encoding.py:62:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:63:13: E128 continuation line under-indented for visual indent
morse_talk/tests/test_encoding.py:64:13: E124 closing bracket does not match visual indentation
morse_talk/tests/test_encoding.py:67:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:73:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_encoding.py:79:9: F405 assert_equal may be undefined, or defined from star imports: nose.tools
morse_talk/tests/test_tree.py:6:1: F401 'morse_talk.tree.MorseBinaryTree' imported but unused
morse_talk/tests/test_tree.py:10:1: E302 expected 2 blank lines, found 1
morse_talk/tests/test_tree.py:14:1: E302 expected 2 blank lines, found 1
morse_talk/tests/test_tree.py:20:1: E302 expected 2 blank lines, found 1
morse_talk/tests/test_tree.py:24:1: E302 expected 2 blank lines, found 1
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,8 +7,8 @@ python:
   - 3.3
   - 3.4
   - 3.5
-  - "pypy"
-  - "pypy3"
+#  - "pypy"
+#  - "pypy3"
 
 cache:
   directories:

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,9 +4,9 @@ language: python
 
 python:
   - 2.7
-  - 3.2
   - 3.3
   - 3.4
+  - 3.5
   - "pypy"
   - "pypy3"
 
@@ -24,11 +24,14 @@ before_install:
  - pip install --upgrade pip
 
 install:
+ - pip install -qq flake8
  - pip install .
 
 script:
   - cd `mktemp -d`
   - nosetests --verbosity=2 --with-doctest
+  - flake8 --version
+  - flake8 --ignore E501 morse_talk
 
 notifications:
   email: false

--- a/.travis.yml
+++ b/.travis.yml
@@ -28,7 +28,6 @@ install:
  - pip install .
 
 script:
-  - cd `mktemp -d`
   - nosetests --verbosity=2 --with-doctest
   - flake8 --version
   - flake8 --ignore E501 morse_talk

--- a/morse_talk/__init__.py
+++ b/morse_talk/__init__.py
@@ -1,5 +1,5 @@
-from .encoding import *
-from .decoding import *
+from .encoding import *  # noqa
+from .decoding import *  # noqa
 
 # Release data
 from morse_talk import release

--- a/morse_talk/cli_mplot.py
+++ b/morse_talk/cli_mplot.py
@@ -41,5 +41,6 @@ def main():
     mplot.plot(message, element_duration)
     plt.show()
 
+
 if __name__ == '__main__':
     main()

--- a/morse_talk/cli_mplot.py
+++ b/morse_talk/cli_mplot.py
@@ -16,10 +16,10 @@ import argparse
 
 import matplotlib.pyplot as plt
 
-import morse_talk as mtalk
 import morse_talk.plot as mplot
 from morse_talk.utils import WORD
 from morse_talk.utils import display, _get_speed
+
 
 def main():
     parser = argparse.ArgumentParser(prog="mplot", description='Plot morse code')
@@ -38,7 +38,7 @@ def main():
 
     display(message, wpm, element_duration, word_ref)
 
-    _ = mplot.plot(message, element_duration)
+    mplot.plot(message, element_duration)
     plt.show()
 
 if __name__ == '__main__':

--- a/morse_talk/cli_msound.py
+++ b/morse_talk/cli_msound.py
@@ -69,5 +69,6 @@ def main():
 
         write_wavefile(filename, samples, samp_nb, args.channels, args.bits // 8, args.rate)
 
+
 if __name__ == "__main__":
     main()

--- a/morse_talk/cli_msound.py
+++ b/morse_talk/cli_msound.py
@@ -19,8 +19,9 @@ from morse_talk.utils import (FRAMERATE, AMPLITUDE, FREQUENCY, WORD)
 from morse_talk.utils import (_get_speed, display, samples_nb)
 
 from morse_talk.sound import (BITS, CHANNELS)
-from morse_talk.sound import (preview_wave,
-                    compute_samples, generate_wave, write_wavefile)
+from morse_talk.sound import (preview_wave, compute_samples,
+                              generate_wave, write_wavefile)
+
 
 def main():
     parser = argparse.ArgumentParser(prog="msound", description='Create or listen morse code sound')
@@ -48,7 +49,7 @@ def main():
 
     display(message, wpm, element_duration, word_ref)
 
-    if args.filename == '': # preview
+    if args.filename == '':  # preview
         preview_wave(message, wpm, args.frequency, framerate, amplitude, word_ref)
         while(args.loop):
             message_with_space = " " + message
@@ -60,10 +61,10 @@ def main():
         samp_nb = samples_nb(message=message, wpm=wpm, framerate=framerate, word_spaced=False)
         samples = compute_samples(channels, samp_nb)
 
-        if args.filename == '-': # write to console
+        if args.filename == '-':  # write to console
             filename = sys.stdout
             write_wavefile(filename, samples, samp_nb, args.channels, args.bits // 8, args.rate)
-        else: # write the samples to a .wav file
+        else:  # write the samples to a .wav file
             filename = args.filename
 
         write_wavefile(filename, samples, samp_nb, args.channels, args.bits // 8, args.rate)

--- a/morse_talk/cli_mtree.py
+++ b/morse_talk/cli_mtree.py
@@ -26,5 +26,6 @@ def main():
     mt = MorseBinaryTree()
     print(mt[c])
 
+
 if __name__ == '__main__':
     main()

--- a/morse_talk/cli_mtree.py
+++ b/morse_talk/cli_mtree.py
@@ -16,6 +16,7 @@ import argparse
 
 from morse_talk.tree import MorseBinaryTree
 
+
 def main():
     parser = argparse.ArgumentParser(description='Display morse binary tree')
     parser.add_argument('-c', '--c', help='Morse code character', default='')

--- a/morse_talk/decoding.py
+++ b/morse_talk/decoding.py
@@ -13,6 +13,7 @@ __all__ = ['decode']
 
 binary_lookup = {encoding.encode(key, encoding_type='binary'): key for key, value in encoding.morsetab.items()}
 
+
 def decode(code, encoding_type='default'):
     """Converts a string of morse code into English message
 
@@ -37,12 +38,12 @@ def decode(code, encoding_type='default'):
         index = {}
 
         for i in range(len(code)):
-            if code[i: i+3] == ' '*3:
-                if code[i: i+7] == ' '*7:
+            if code[i:i + 3] == ' ' * 3:
+                if code[i:i + 7] == ' ' * 7:
                     words += 1
                     letters += 1
                     index[words] = letters
-                elif code[i+4] and code[i-1] != ' ':  # Check for '   '
+                elif code[i + 4] and code[i - 1] != ' ':  # Check for '   '
                     letters += 1
 
         message = [reversed_morsetab[i] for i in code.split()]
@@ -51,7 +52,7 @@ def decode(code, encoding_type='default'):
         return ''.join(message)
 
     elif encoding_type == 'binary':
-        lst = list(map(lambda word: word.split('0'*3), code.split('0'*7)))
+        lst = list(map(lambda word: word.split('0' * 3), code.split('0' * 7)))
         # list of list of character (each sub list being a word)
         for i, word in enumerate(lst):
             for j, bin_letter in enumerate(word):

--- a/morse_talk/encoding.py
+++ b/morse_talk/encoding.py
@@ -7,9 +7,9 @@ Functions to encode strings
 #    All rights reserved.
 #    GNU GPL v2 license.
 
-__all__ = ['encode']
-
 import collections
+
+__all__ = ['encode']
 
 morsetab = collections.OrderedDict([
     ('A', '.-'),
@@ -62,6 +62,7 @@ morsetab = collections.OrderedDict([
     ('_', '..--.-')
 ])
 
+
 def _split_message(message):
     """
     >>> _split_message("SOS SOS")
@@ -76,6 +77,7 @@ def _split_message(message):
     word_sep = " "
     return list(map(list, message.split(word_sep)))
 
+
 def _encode_morse(message):
     """
     >>> message = "SOS"
@@ -86,6 +88,7 @@ def _encode_morse(message):
     [' ', '...', '---', '...']
     """
     return [morsetab.get(c.upper(), '?') for c in message]
+
 
 def _encode_to_morse_string(message, letter_sep):
     """
@@ -103,6 +106,7 @@ def _encode_to_morse_string(message, letter_sep):
         return s
     return letter_sep.join([to_string(i, s) for i, s in enumerate(_encode_morse(message))])
 
+
 def _encode_binary(message, on=1, off=0):
     """
     >>> message = "SOS"
@@ -115,10 +119,11 @@ def _encode_binary(message, on=1, off=0):
     l = _encode_morse(message)
     s = ' '.join(l)
     l = list(s)
-    bin_conv = { '.': [on], '-': [on]*3, ' ': [off]}
+    bin_conv = {'.': [on], '-': [on] * 3, ' ': [off]}
     l = map(lambda symb: [off] + bin_conv[symb], l)
-    lst = [item for sublist in l for item in sublist] # flatten list
+    lst = [item for sublist in l for item in sublist]  # flatten list
     return lst[1:]
+
 
 def _encode_to_binary_string(message, on, off):
     """
@@ -136,7 +141,8 @@ def _encode_to_binary_string(message, on, off):
         return s
     return ''.join(to_string(i, s) for i, s in enumerate(_encode_binary(message, on=on, off=off)))
 
-def encode(message, encoding_type='default', letter_sep = ' '*3, strip=True):
+
+def encode(message, encoding_type='default', letter_sep=' ' * 3, strip=True):
     """Converts a string of message into morse
 
     Two types of marks are there. One is short mark, dot(.) or "dit" and
@@ -184,9 +190,11 @@ def encode(message, encoding_type='default', letter_sep = ' '*3, strip=True):
     else:
         raise NotImplementedError("encoding_type must be in %s" % allowed_encoding_type)
 
+
 def main():
     import doctest
     doctest.testmod()
+
 
 if __name__ == '__main__':
     main()

--- a/morse_talk/plot.py
+++ b/morse_talk/plot.py
@@ -13,6 +13,7 @@ import matplotlib.pyplot as plt
 
 from morse_talk.encoding import _encode_binary
 
+
 def _create_ax(ax):
     """
     Create a Matplotlib Axe from ax
@@ -25,6 +26,7 @@ def _create_ax(ax):
         return axs
     else:
         return ax
+
 
 def _create_x_y(l, duration=1):
     """
@@ -50,7 +52,8 @@ def _create_x_y(l, duration=1):
         x.append(i * duration)
     return x, y
 
-def plot(message, duration=1, ax = None):
+
+def plot(message, duration=1, ax=None):
     """
     Plot a message
 
@@ -67,9 +70,11 @@ def plot(message, duration=1, ax = None):
     ax.set_xlim(-delta_x, len(lst_bin) * duration + delta_x)
     return ax
 
+
 def main():
     import doctest
     doctest.testmod()
+
 
 if __name__ == '__main__':
     main()

--- a/morse_talk/sound.py
+++ b/morse_talk/sound.py
@@ -31,10 +31,12 @@ try:
 except AttributeError:
     stdout = sys.stdout
 
+
 def grouper(n, iterable, fillvalue=None):
     "grouper(3, 'ABCDEFG', 'x') --> ABC DEF Gxx"
     args = [iter(iterable)] * n
     return zip_longest(fillvalue=fillvalue, *args)
+
 
 def compute_samples(channels, nsamples=None):
     '''
@@ -43,6 +45,7 @@ def compute_samples(channels, nsamples=None):
     at each sample in the file for each channel.
     '''
     return islice(izip(*(imap(sum, izip(*channel)) for channel in channels)), nsamples)
+
 
 def write_wavefile(f, samples, nframes=None, nchannels=2, sampwidth=2, framerate=44100, bufsize=2048):
     "Write samples to a wavefile."
@@ -58,8 +61,9 @@ def write_wavefile(f, samples, nframes=None, nchannels=2, sampwidth=2, framerate
     for chunk in grouper(bufsize, samples):
         frames = b''.join(b''.join(struct.pack('h', int(max_amplitude * sample)) for sample in channels) for channels in chunk if channels is not None)
         w.writeframesraw(frames)
-    
+
     w.close()
+
 
 def sine_wave(i, frequency=FREQUENCY, framerate=FRAMERATE, amplitude=AMPLITUDE):
     """
@@ -89,13 +93,16 @@ def generate_wave(message, wpm=WPM, framerate=FRAMERATE, skip_frame=0, amplitude
 
     """
     lst_bin = _encode_binary(message)
-    if amplitude > 1.0: amplitude = 1.0
-    if amplitude < 0.0: amplitude = 0.0
-    seconds_per_dot = _seconds_per_dot(word_ref) # =1.2
+    if amplitude > 1.0:
+        amplitude = 1.0
+    if amplitude < 0.0:
+        amplitude = 0.0
+    seconds_per_dot = _seconds_per_dot(word_ref)  # =1.2
     for i in count(skip_frame):
         bit = morse_bin(i=i, lst_bin=lst_bin, wpm=wpm, framerate=framerate, default_value=0.0, seconds_per_dot=seconds_per_dot)
         sine = sine_wave(i=i, frequency=frequency, framerate=framerate, amplitude=amplitude)
         yield sine * bit
+
 
 def morse_bin(i, lst_bin, wpm=WPM, framerate=FRAMERATE, default_value=0.0, seconds_per_dot=1.2):
     """
@@ -105,16 +112,18 @@ def morse_bin(i, lst_bin, wpm=WPM, framerate=FRAMERATE, default_value=0.0, secon
     try:
         return lst_bin[int(float(wpm) * float(i) / (seconds_per_dot * float(framerate)))]
     except IndexError:
-        return default_value   
+        return default_value
+
 
 def calculate_wave(i, lst_bin, wpm=WPM, frequency=FREQUENCY, framerate=FRAMERATE, amplitude=AMPLITUDE, seconds_per_dot=SECONDS_PER_DOT):
     """
     Returns product of a sin wave and morse code (dit, dah, silent)
     """
-    bit = morse_bin(i=i, lst_bin=lst_bin, wpm=wpm, framerate=framerate, 
-            default_value=0.0, seconds_per_dot=seconds_per_dot)
+    bit = morse_bin(i=i, lst_bin=lst_bin, wpm=wpm, framerate=framerate,
+                    default_value=0.0, seconds_per_dot=seconds_per_dot)
     sine = sine_wave(i=i, frequency=frequency, framerate=framerate, amplitude=amplitude)
     return bit * sine
+
 
 def preview_wave(message, wpm=WPM, frequency=FREQUENCY, framerate=FRAMERATE, amplitude=AMPLITUDE, word_ref=WORD):
     """
@@ -129,12 +138,14 @@ def preview_wave(message, wpm=WPM, frequency=FREQUENCY, framerate=FRAMERATE, amp
     amplitude = _limit_value(amplitude)
     seconds_per_dot = _seconds_per_dot(word_ref)  # 1.2
     a = [calculate_wave(i, lst_bin, wpm, frequency, framerate, amplitude, seconds_per_dot)
-        for i in range(samp_nb)]
+         for i in range(samp_nb)]
     sd.play(a, framerate, blocking=True)
+
 
 def main():
     import doctest
     doctest.testmod()
+
 
 if __name__ == '__main__':
     main()

--- a/morse_talk/tests/test_encoding.py
+++ b/morse_talk/tests/test_encoding.py
@@ -1,6 +1,8 @@
 #!/usr/bin/env python
-from nose.tools import *
+from nose.tools import *  # noqa
+from nose.tools import assert_equal
 import morse_talk as mtalk
+
 
 class TestEncoding:
 
@@ -9,9 +11,8 @@ class TestEncoding:
         message = message.upper()
         morse_code = mtalk.encode(message)
         morse_code_expected = ('-   ....   ..   ...       ..   ...       .-       -   .'
-            '   ...   -       ..-.   ---   .-.       -..   .   ..-.   .-   ..- '
-            '  .-..   -       .   -.   -.-.   ---   -..   ..   -.   --.   .-.-.-'
-            )
+                               '   ...   -       ..-.   ---   .-.       -..   .   ..-.   .-   ..- '
+                               '  .-..   -       .   -.   -.-.   ---   -..   ..   -.   --.   .-.-.-')
         assert_equal(morse_code, morse_code_expected)
 
     def test_encoding_binary(self):
@@ -19,11 +20,9 @@ class TestEncoding:
         message = message.upper()
         morse_code = mtalk.encode(message)
         morse_code_expected = ('-   ....   ..   ...       ..   ...       .-       -   .'
-            '   ...   -       ..-.   ---   .-.       -...   ..   -.   .-   .-.'
-            '   -.--       .   -.   -.-.   ---   -..   ..   -.   --.   .-.-.-'
-            )
+                               '   ...   -       ..-.   ---   .-.       -...   ..   -.   .-   .-.'
+                               '   -.--       .   -.   -.-.   ---   -..   ..   -.   --.   .-.-.-')
         assert_equal(morse_code, morse_code_expected)
-
 
     def test_encode_morse1(self):
         message = 'MORSE CODE'
@@ -35,33 +34,33 @@ class TestEncoding:
         message = 'Alpha Ranger 45 departed'
         morse_code = mtalk.encode(message)
         morse_code_expected = (".-   .-..   .--.   ....   .-       "
-                ".-.   .-   -.   --.   .   .-.       ....-   .....       -..   .   .--.   .-   .-.   -   .   -..")
+                               ".-.   .-   -.   --.   .   .-.       ....-   .....       -..   .   .--.   .-   .-.   -   .   -..")
         assert_equal(morse_code, morse_code_expected)
 
     def test_encode_binary(self):
         message = 'Alpha Ranger 45 knocked down'
         binary = mtalk.encode(message, encoding_type='binary')
         binary_expected = ('10111000101110101000101110111010001010101000101110000000'
-            '10111010001011100011101000111011101000100010111010000000'
-            '10101010111000101010101000000011101011100011101000111011'
-            '10111000111010111010001110101110001000111010100000001110'
-            '1010001110111011100010111011100011101')
+                           '10111010001011100011101000111011101000100010111010000000'
+                           '10101010111000101010101000000011101011100011101000111011'
+                           '10111000111010111010001110101110001000111010100000001110'
+                           '1010001110111011100010111011100011101')
         assert_equal(binary, binary_expected)
 
     def test_encode_binary2(self):
         message = 'MORSE CODE'
         binary = mtalk.encode(message, encoding_type='binary')
         binary_expected = ('11101110001110111011100010111010001010100010000000'
-            '111010111010001110111011100011101010001')
+                           '111010111010001110111011100011101010001')
         assert_equal(binary, binary_expected)
+
 
 class TestDecoding:
 
     def test_decoding_default(self):
         code = ('-   ....   ..   ...       ..   ...       .-       -   .'
-            '   ...   -       ..-.   ---   .-.       -..   .   ..-.   .-   ..- '
-            '  .-..   -       .   -.   -.-.   ---   -..   ..   -.   --.   .-.-.-'
-            )
+                '   ...   -       ..-.   ---   .-.       -..   .   ..-.   .-   ..- '
+                '  .-..   -       .   -.   -.-.   ---   -..   ..   -.   --.   .-.-.-')
         message = mtalk.decode(code)
         message_expected = 'THIS IS A TEST FOR DEFAULT ENCODING.'
         assert_equal(message, message_expected)

--- a/morse_talk/tests/test_tree.py
+++ b/morse_talk/tests/test_tree.py
@@ -2,14 +2,15 @@
 # -*- coding: utf-8 -*-
 
 from nose.tools import assert_equals
-import morse_talk as mtalk
 from morse_talk.tree import MorseBinaryTree
 
-mt = mtalk.tree.MorseBinaryTree()
+mt = MorseBinaryTree()
+
 
 def test_tree_pretty():
     print(mt)
     # printing tree shouldn't raises exception
+
 
 def test_tree_getitem_len1():
     assert_equals(mt['.'].char, 'E')
@@ -17,9 +18,11 @@ def test_tree_getitem_len1():
     assert_equals(mt['.']['.']['.'].char, 'S')
     assert_equals(mt['-']['-']['-'].char, 'O')
 
+
 def test_tree_properties_dit_dah():
     assert_equals(mt.dit.dit.dit.char, 'S')
     assert_equals(mt.dah.dah.dah.char, 'O')
+
 
 def test_tree_getitem_len3():
     assert_equals(mt['...'].char, 'S')

--- a/morse_talk/tree.py
+++ b/morse_talk/tree.py
@@ -32,6 +32,7 @@ print(mt['...'])
 
 import morse_talk as mtalk
 
+
 class MorseBinaryTree(object):
     """
     Morse binary tree
@@ -65,13 +66,13 @@ class MorseBinaryTree(object):
 
     def __str__(self):
         return self._pretty_pprint()
-        #return self._pretty_custom(self._tree)
+        # return self._pretty_custom(self._tree)
 
     def _pretty_pprint(self):
         import pprint
         return pprint.pformat(self._tree)
-    
-    def _pretty_custom(self, d, indent_nb = 0, indent_char = '  '):
+
+    def _pretty_custom(self, d, indent_nb=0, indent_char='  '):
         s = ''
         for key, value in d.items():
             if key != self._char:
@@ -79,9 +80,9 @@ class MorseBinaryTree(object):
             if isinstance(value, dict):
                 s += self._pretty_custom(value, indent_nb + 1)
             else:
-                #s += ((indent_char * (indent_nb + 1) + str(value))) + '\n'
+                # s += ((indent_char * (indent_nb + 1) + str(value))) + '\n'
                 s += "%20s\n" % value
-            #s += '\n'
+            # s += '\n'
         return s
 
     @property
@@ -109,9 +110,11 @@ class MorseBinaryTree(object):
             mt = MorseBinaryTree(tree=tree)[morse_code[1:]]
             return mt
 
+
 def main():
     import doctest
     doctest.testmod()
+
 
 if __name__ == '__main__':
     main()

--- a/morse_talk/utils.py
+++ b/morse_talk/utils.py
@@ -10,6 +10,11 @@ All rights reserved.
 
 """
 
+from enum import Enum
+
+import morse_talk as mtalk
+from morse_talk.encoding import (_split_message, _encode_binary, _encode_to_binary_string)
+
 WORD = 'PARIS'  # Reference word for code speed
 # http://www.kent-engineers.com/codespeed.htm
 
@@ -18,10 +23,6 @@ FRAMERATE = 44100 / 4  # default framerate
 FREQUENCY = 750  # default sound frequency
 AMPLITUDE = 0.5
 
-from enum import Enum
-
-import morse_talk as mtalk
-from morse_talk.encoding import (_split_message, _encode_binary, _encode_to_binary_string)
 
 def _repeat_word(word, N, word_space=" "):
     """
@@ -39,6 +40,7 @@ def _repeat_word(word, N, word_space=" "):
     message = message[len(word_space):]
     return message
 
+
 def mlength(message, N=1, word_spaced=True):
     """
     Returns Morse length
@@ -55,12 +57,13 @@ def mlength(message, N=1, word_spaced=True):
     lst_bin = _encode_binary(message)
     N = len(lst_bin)
     if word_spaced:
-        N -= 1 # E is one "dit" so we remove it
+        N -= 1  # E is one "dit" so we remove it
     return N
+
 
 def wpm_to_duration(wpm, output='timedelta', word_ref=WORD):
     """
-    Convert from WPM (word per minutes) to 
+    Convert from WPM (word per minutes) to
     element duration
 
     Parameters
@@ -109,9 +112,10 @@ def wpm_to_duration(wpm, output='timedelta', word_ref=WORD):
         raise NotImplementedError("output must be in %s" % allowed_output)
     return duration
 
+
 def duration(message, wpm, output='timedelta', word_ref=WORD, word_spaced=False):
     """
-    Calculate duration to send a message at a given code speed 
+    Calculate duration to send a message at a given code speed
     (calculated using reference word PARIS)
 
     Parameters
@@ -142,6 +146,7 @@ def duration(message, wpm, output='timedelta', word_ref=WORD, word_spaced=False)
     word_length = mlength(message, word_spaced=word_spaced)
     return word_length * elt_duration
 
+
 def samples_nb(message, wpm, framerate=FRAMERATE, word_spaced=False):
     """
     Calculate the number of samples for a given word at a given framerate (samples / seconds)
@@ -151,18 +156,20 @@ def samples_nb(message, wpm, framerate=FRAMERATE, word_spaced=False):
     """
     return int(duration(message, wpm, output='float', word_spaced=word_spaced) / 1000.0 * framerate)
 
+
 def _seconds_per_dot(word_ref=WORD):
     """
     >>> _seconds_per_dot('PARIS')
     1.2
     """
-    return 60 / mlength(word_ref) # 1.2 with 'PARIS'
+    return 60 / mlength(word_ref)  # 1.2 with 'PARIS'
+
 
 def _get_speed(element_duration, wpm, word_ref=WORD):
     """
     Returns
         element duration when element_duration and/or code speed is given
-        wpm    
+        wpm
 
     >>> _get_speed(0.2, None)
     (0.2, 5.999999999999999)
@@ -175,19 +182,20 @@ def _get_speed(element_duration, wpm, word_ref=WORD):
     """
     seconds_per_dot = _seconds_per_dot(word_ref)
     if element_duration is None and wpm is None:
-        #element_duration = 1
-        #wpm = seconds_per_dot / element_duration
+        # element_duration = 1
+        # wpm = seconds_per_dot / element_duration
         wpm = WPM
-        element_duration= wpm_to_duration(wpm, output='float', word_ref=WORD) / 1000.0
+        element_duration = wpm_to_duration(wpm, output='float', word_ref=WORD) / 1000.0
         return element_duration, wpm
     elif element_duration is not None and wpm is None:
         wpm = seconds_per_dot / element_duration
         return element_duration, wpm
     elif element_duration is None and wpm is not None:
-        element_duration= wpm_to_duration(wpm, output='float', word_ref=WORD) / 1000.0
+        element_duration = wpm_to_duration(wpm, output='float', word_ref=WORD) / 1000.0
         return element_duration, wpm
     else:
         raise NotImplementedError("Can't set both element_duration and wpm")
+
 
 def _numbers_units(N):
     """
@@ -196,6 +204,7 @@ def _numbers_units(N):
     """
     lst = range(1, N + 1)
     return "".join(list(map(lambda i: str(i % 10), lst)))
+
 
 def _numbers_decades(N):
     """
@@ -206,6 +215,7 @@ def _numbers_decades(N):
     lst = range(1, N + 1)
     return "".join(map(lambda i: "%10s" % i, lst))
 
+
 class ALIGN(Enum):
     """
     An Enum class for text alignement
@@ -213,6 +223,7 @@ class ALIGN(Enum):
     LEFT = 0
     CENTER = 1
     RIGHT = 2
+
 
 def _char_to_string_binary(c, align=ALIGN.LEFT, padding='-'):
     """
@@ -238,6 +249,7 @@ def _char_to_string_binary(c, align=ALIGN.LEFT, padding='-'):
     s = "{0:" + padding + s_align + str(N) + "}"
     return s.format(c)
 
+
 def _timing_representation(message):
     """
     Returns timing representation of a message like
@@ -260,6 +272,7 @@ def _timing_representation(message):
     s += '\n' + _timing_char(message)
     return s
 
+
 def _timing_char(message):
     """
     >>> message = 'MORSE CODE'
@@ -279,6 +292,7 @@ def _timing_char(message):
             s += _char_to_string_binary(c, align=ALIGN.LEFT)
     return s
 
+
 def _char_to_string_morse(c, align=ALIGN.CENTER, padding=' '):
     """
     >>> _char_to_string_morse('O')
@@ -296,6 +310,7 @@ def _char_to_string_morse(c, align=ALIGN.CENTER, padding=' '):
         raise NotImplementedError("align '%s' not allowed" % align)
     s = "{0:" + padding + s_align + str(N) + "}"
     return s.format(c)
+
 
 def _spoken_representation_L1(lst_lst_char):
     """
@@ -319,6 +334,7 @@ def _spoken_representation_L1(lst_lst_char):
             s += _char_to_string_morse(c)
     return s
 
+
 def _spoken_representation_L2(lst_lst_char):
     """
     >>> lst = [['M', 'O', 'R', 'S', 'E'], ['C', 'O', 'D', 'E']]
@@ -337,6 +353,7 @@ def _spoken_representation_L2(lst_lst_char):
             s += mtalk.encoding.morsetab[c]
     return s
 
+
 def _spoken_representation(message):
     """
     Returns 2 lines of spoken representation of a message
@@ -350,9 +367,10 @@ def _spoken_representation(message):
     s += '\n' + _spoken_representation_L2(lst_lst_char)
     return s
 
+
 def display(message, wpm, element_duration, word_ref, strip=False):
     """
-    Display 
+    Display
         text message
         morse code
         binary morse code
@@ -377,10 +395,11 @@ def display(message, wpm, element_duration, word_ref, strip=False):
     print("reference word : %r" % word_ref)
     print("")
 
+
 def _limit_value(value, upper=1.0, lower=0.0):
     """
     Returs value (such as amplitude) to upper and lower value
-    
+
     >>> _limit_value(0.5)
     0.5
 
@@ -390,15 +409,19 @@ def _limit_value(value, upper=1.0, lower=0.0):
     >>> _limit_value(-1.5)
     0.0
     """
-    if value > upper: return(upper)
-    if value < lower: return(lower)
+    if value > upper:
+        return(upper)
+    if value < lower:
+        return(lower)
     return value
 
 SECONDS_PER_DOT = _seconds_per_dot(WORD)  # 1.2
 
+
 def main():
     import doctest
     doctest.testmod()
+
 
 if __name__ == '__main__':
     main()

--- a/morse_talk/utils.py
+++ b/morse_talk/utils.py
@@ -162,7 +162,7 @@ def _seconds_per_dot(word_ref=WORD):
     >>> _seconds_per_dot('PARIS')
     1.2
     """
-    return 60 / mlength(word_ref)  # 1.2 with 'PARIS'
+    return 60.0 / mlength(word_ref)  # 1.2 with 'PARIS'
 
 
 def _get_speed(element_duration, wpm, word_ref=WORD):

--- a/morse_talk/utils.py
+++ b/morse_talk/utils.py
@@ -415,6 +415,7 @@ def _limit_value(value, upper=1.0, lower=0.0):
         return(lower)
     return value
 
+
 SECONDS_PER_DOT = _seconds_per_dot(WORD)  # 1.2
 
 

--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,7 @@ setup(
     packages=[
         'morse_talk'
     ],
-    install_requires=['sounddevice'],
+    install_requires=['sounddevice', 'matplotlib'],
     entry_points={
         'console_scripts': [
             'mplot=morse_talk.cli_mplot:main',

